src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/UploadEnvelopeDocumentsTask.java

```diff
@@ -35,6 +42,8 @@ public UploadEnvelopeDocumentsTask(
         this.uploadService = uploadService;
     }
 
+    @Scheduled(fixedDelayString = "${scheduling.task." + TASK_NAME + ".delay}")
+    @SchedulerLock(name = TASK_NAME) // so to not upload documents multiple times
```

Comment:
Should the task name be the name of the reupload task until it's gone?

Reply:
I just realised these two jobs look for envelopes in different statuses. All good.

Reply:
yep. about to submit a pr for reupload complete re-write ;)